tests/test_config_reader.py

```diff
@@ -14,3 +14,8 @@ def test_parse_config_files():
         config_data = parse_config_file("Not_Real_Directory")
     with pytest.raises(SystemExit):
         config_data = parse_config_file(THIS_DIR)
+    with pytest.raises(SystemExit) as pytest_exit_object:
+        config_data = parse_config_file("Not_Real_Directory")
+        assert pytest_exit_object.type == SystemExit 
+        assert pytest_exit_object.value.code == 42  
+        assert pytest_exit_object.sys.stdout.getline().strip() == "Could not find directory or config. Exiting..."
```

Comment:
I would make two different error messages (to make it easier for users).  I would also pass back to them the directory that the code was looking in for the config file. something like:
1) "directory containing config file, {}, not found. Exiting".format(dir)
2) "no file named config.yml found in config directory {}".format(dir)